src/main/java/org/thoughtcrime/securesms/connect/DcEventCenter.java

```diff
@@ -171,7 +171,11 @@ public long handleEvent(@NonNull DcEvent event) {
 
     switch (id) {
       case DcContext.DC_EVENT_INCOMING_MSG:
-        DcHelper.getNotificationCenter(context).addNotification(accountId, event.getData1Int(), event.getData2Int());
+        DcHelper.getNotificationCenter(context).notifyMessage(accountId, event.getData1Int(), event.getData2Int());
+        break;
+
+      case DcContext.DC_EVENT_INCOMING_REACTION:
+        DcHelper.getNotificationCenter(context).notifyReaction(accountId, event.getData1Int(), event.getData2Int(), event.getData2Str());
```

Comment:
one problem with this is that if user "reads"/see  the reaction in one chat, will it be marked as read and the notification disappear in other devices??? or will the user need to handle duplicated notifications in every device?

Reply:
good question!

Reply:
The `MsgsNoticed` contains a chat id (not a message id), so we just need to make sure that core sends it not only when messages are noticed but also when reactions are noticed on another device